### PR TITLE
Add convert_trial() to Subscription

### PR DESCRIPTION
**Pull request by joannasese**
To convert trial to paid subscription with 3d secure action result token id:
```python
subscription.convert_trial(token)
```

To convert trial for non-3ds where subscription has billing info:
```python
subscription.convert_trial()
```

To convert trial to paid subscription with transaction type:
```python
subscription.convert_trial_moto()
```

### CODE REVIEW

File: recurly/__init__.py [hunk 1]
```diff
@@ -1267,25 +1267,54 @@ def convert_trial_moto(self):
...
+            self.raise_http_error(response)
+
+        self.update_from_element(ElementTree.fromstring(response.read()))
+
+
+    # def convert_trial(self, three_d_secure_action_result_token_id = None):
+    #     """Convert trial to paid subscription"""
+    #     account = recurly.Account()
+    #     billing_info = recurly.BillingInfo(
+    #         three_d_secure_action_result_token_id = three_d_secure_action_result_token_id
```
Comment (bhelx) [inline recurly/__init__.py]: I think you should skip using Resources entirely and just build the xml directly. Example:

```python
request = ElementTreeBuilder.Element('subscription')
token_el = ElementTreeBuilder.SubElement(request, 'three_d_secure_action_result_token_id')
token_el.text = 'mytoken'
body = ElementTree.tostring(elem, encoding='UTF-8')
# => <subscription><three_d_secure_action_result_token_id>mytoken</three_d_secure_action_result_token_id></subscription>
```

You should be able to use the same technique to build the other request.

File: recurly/__init__.py [hunk 2]
```diff
@@ -46,7 +46,7 @@
 API_KEY = None
 """The API key to use when authenticating API requests."""
 
-API_VERSION = '2.25'
+API_VERSION = '2.99'
```
Comment (bhelx) [inline recurly/__init__.py]: Reminder: set this back to 2.25 before merging.

File: recurly/__init__.py [hunk 3]
```diff
@@ -1250,6 +1250,71 @@ def resume(self):
...
+        if response.status not in (200, 201, 204):
+            self.raise_http_error(response)
+
+        self.update_from_element(ElementTree.fromstring(response.read()))
+
+    def convert_trial(self, three_d_secure_action_result_token_id = None):
+        """Convert trial to paid subscription"""
+        url = urljoin(self._url, '/convert_trial')
+        
+        if not three_d_secure_action_result_token_id == None:
```
Comment (bhelx) [inline recurly/__init__.py]: I'd change to:

```python
body = None
if three_d_secure_action_result_token_id:
    # build body
    body = ElementTree.tostring(elem, encoding='UTF-8') 
```

File: tests/test_resources.py [hunk 4]
```diff
@@ -1383,6 +1383,21 @@ def test_subscription_pause_resume(self):
         with self.mock_request('subscription/resume.xml'):
             sub.resume()
 
+    def test_subscription_convert_trial(self):
+        billing_info = BillingInfo()
+        billing_info.three_d_secure_action_result_token_id = 'token'
+
+        with self.mock_request('subscription/show-trial.xml'):
+            sub = Subscription.get('123456789012345678901234567890ab')
+
```
Comment (joannasese) [inline tests/test_resources.py]: Missing a test for `sub.convert_trial()` at the moment.

**Pull request merged**